src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -1386,6 +1387,31 @@ internal override IEnumerable<FieldSymbol> GetFieldsToEmit()
                             yield return associatedField;
                         }
                         break;
+                    case SymbolKind.Property:
+                        if (m is SourcePropertySymbolBase propertySymbol)
+                        {
+                            // Ensure the binding is done so we guarantee that we have the BackingField set if necessary.
+                            if (propertySymbol.BackingField is null &&
```

Comment:
>propertySymbol.BackingField is null && [](http://example.com/codeflow?start=32&length=38)

We probably don't want to repeat the binding process over and over again for scenarios that do not require us to synthesize the field after all.

Reply:
I'm not sure if the way I fixed this is good enough.

Reply:
I'm interpreting this comment the same as https://github.com/dotnet/roslyn/pull/57076#discussion_r745881552.